// next-sanity/dist/webhook/parseBody.cjs
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
var webhook = require('@sanity/webhook');
var readBody = require('./readBody.cjs');
async function parseBody(req, secret) {
  let waitForContentLakeEventualConsistency = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : true;
  if (typeof EdgeRuntime !== "undefined") {
    throw new TypeError("The edge runtime isn't supported. You'll have to use the 'nodejs' runtime until the underlying `@sanity/webhook` package is updated to support it.");
  }
  return "text" in req ? parseAppBody(req, secret, waitForContentLakeEventualConsistency) : parsePageBody(req, secret, waitForContentLakeEventualConsistency);
}
async function parsePageBody(req, secret) {
  let waitForContentLakeEventualConsistency = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : true;
  let signature = req.headers[webhook.SIGNATURE_HEADER_NAME];
  if (Array.isArray(signature)) {
    signature = signature[0];
  }
  if (!signature) {
    console.error("Missing signature header");
    return {
      body: null,
      isValidSignature: null
    };
  }
  if (req.readableEnded) {
    throw new Error("Request already ended and the POST body can't be read. Have you setup `export {config} from 'next-sanity/webhook' in your webhook API handler?`");
  }
  const body = await readBody._readBody(req);
  const validSignature = secret ? webhook.isValidSignature(body, signature, secret.trim()) : null;
  if (validSignature !== false && waitForContentLakeEventualConsistency) {
    await new Promise(resolve => setTimeout(resolve, 1e3));
  }
  return {
    body: body.trim() ? JSON.parse(body) : null,
    isValidSignature: validSignature
  };
}
async function parseAppBody(req, secret) {
  let waitForContentLakeEventualConsistency = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : true;
  const signature = req.headers.get(webhook.SIGNATURE_HEADER_NAME);
  if (!signature) {
    console.error("Missing signature header");
    return {
      body: null,
      isValidSignature: null
    };
  }
  const body = await req.text();
  const validSignature = secret ? webhook.isValidSignature(body, signature, secret.trim()) : null;
  if (validSignature !== false && waitForContentLakeEventualConsistency) {
    await new Promise(resolve => setTimeout(resolve, 1e3));
  }
  return {
    body: body.trim() ? JSON.parse(body) : null,
    isValidSignature: validSignature
  };
}
exports.parseAppBody = parseAppBody;
exports.parseBody = parseBody;
